packages/trader/src/App/Containers/Modals/modals.jsx

```diff
@@ -1,90 +1,110 @@
-import React       from 'react';
-import { connect } from 'Stores/connect';
-import { urlFor }  from '_common/url';
-import 'Sass/app/modules/modals.scss';
+import PropTypes                from 'prop-types';
+import React, { Component } from 'react';
+import { connect }              from 'Stores/connect';
+import { urlFor }               from '_common/url';
 import UnsupportedContractModal from 'App/Components/Elements/Modals/UnsupportedContractModal';
 import MarketUnavailableModal   from 'App/Components/Elements/Modals/MarketUnavailableModal';
 import DenialOfServiceModal     from 'App/Components/Elements/Modals/DenialOfServiceModal';
 import ServicesErrorModal       from 'App/Components/Elements/Modals/ServicesErrorModal';
+import 'Sass/app/modules/modals.scss';
+import { deeperCompare } from '../../../Utils/React/deeper-compare';
 
 // const AccountSignupModal       = React.lazy(() => import(/* webpackChunkName: "AccountSignupModal" */'./Containers/AccountSignupModal'));
 
-const Modals = ({
-    clearPurchaseInfo,
-    is_denial_of_service_modal_visible,
-    is_unsupported_contract_modal_visible,
-    is_market_unavailable_visible,
-    is_services_error_visible,
-    toggleUnsupportedContractModal,
-    setHasOnlyForwardingContracts,
-    resetPreviousSymbol,
-    toggleServicesErrorModal,
-    resetPurchase,
-    services_error,
-    switchAccount,
-    virtual_account_loginid,
-}) => {
-    const denialOfServiceOnCancel = () => {
+class Modals extends Component {
+    shouldComponentUpdate (next_props, next_state) {
```

Comment:
hmm doesn't mobx handle this? since it's a connected component

Reply:
Yes it does. I'll update the code. thanks.